src/main/java/software/amazon/awssdk/crt/http/HttpRequest.java

```diff
@@ -123,22 +125,21 @@ public HttpRequestBodyStream getBodyStream() {
     /**
      * Requests are marshalled as follows:
      *
-     * each string field is:
-     * [4-bytes BE] [variable length bytes specified by the previous field]
+     * each string field is: [4-bytes BE] [variable length bytes specified by the
+     * previous field]
      *
-     * Each request is then:
-     * [method][path][header name-value pairs]
+     * Each request is then: [method][path][header name-value pairs]
+     * 
      * @return encoded blob of headers
      */
     public byte[] marshalForJni() {
         int size = 0;
         size += BUFFER_INT_SIZE + method.length();
         size += BUFFER_INT_SIZE + encodedPath.length();
-        size += (BUFFER_INT_SIZE * 2) * headers.size();
 
-        for(HttpHeader header : headers) {
+        for (HttpHeader header : headers) {
             if (header.getNameBytes().length > 0) {
```

Comment:
That feels weird... We silently rule out the empty headers for users, which I don't think we should do... We should just let it pass by and fail hard from either http level or the other side.

So, maybe just remove this check???

Reply:
This is from an old PR, mentioned you in relevant slack conversation. Just fixed a bug where headers size was incorrectly calculated.